srai/loaders/osm_loaders/filters/_typing.py

```diff
@@ -8,6 +9,34 @@
 GroupedOsmTagsFilter = dict[str, OsmTagsFilter]
 
 
+def merge_osm_tags_filter(tags: Union[OsmTagsFilter, GroupedOsmTagsFilter]) -> OsmTagsFilter:
+    """
+    Merge OSM tags filter into `osm_tags_type` type.
+
+    Optionally merges `GroupedOsmTagsFilter` into `OsmTagsFilter` to allow loaders to load all
+    defined groups during single operation.
+
+    Args:
+        tags (Union[OsmTagsFilter, GroupedOsmTagsFilter]): OSM tags filter definition.
+
+    Raises:
+        AttributeError: When provided tags don't match both
+            `OsmTagsFilter` or `GroupedOsmTagsFilter`.
+
+    Returns:
+        osm_tags_type: Merged filters.
+    """
+    if is_expected_type(tags, OsmTagsFilter):
+        return cast(OsmTagsFilter, tags)
+    elif is_expected_type(tags, GroupedOsmTagsFilter):
+        return merge_grouped_osm_tags_filter(cast(GroupedOsmTagsFilter, tags))
+
+    raise AttributeError(
+        "Provided tags don't match required type definitions"
+        " (OsmTagsFilter or GroupedOsmTagsFilter)."
+    )
+
+
 def merge_grouped_osm_tags_filter(grouped_filter: GroupedOsmTagsFilter) -> OsmTagsFilter:
```

Comment:
Should we maybe make this private to simplify the API since the function above handles this?

Reply:
That's a good idea. I've even expanded the functionality of the function.